Source/Core/Core/WiiRoot.cpp

```diff
@@ -142,32 +142,28 @@ void InitializeWiiFileSystemContents()
   SysConf sysconf{fs};
   sysconf.Save();
 
-  InitializeDeterministicWiiSaves();
+  InitializeDeterministicWiiSaves(fs.get());
 }
 
 void CleanUpWiiFileSystemContents()
 {
-  if (s_temp_wii_root.empty())
+  if (s_temp_wii_root.empty() || !SConfig::GetInstance().bEnableMemcardSdWriting)
```

Comment:
Looks like a bug if this was missing before, right?

Reply:
It wasn't missing before, but somewhat hard to see as the check was put immediately after tons of unrelated stuff with no vertical spacing.

I thought the nested levels of indentation were not very readable so I just changed the function to return earlier.